src/serac/physics/heat_transfer.hpp

```diff
@@ -686,24 +761,35 @@ class HeatTransfer<order, dim, Parameters<parameter_space...>, std::integer_sequ
 
   /**
    * @brief Compute the implicit sensitivity of the quantity of interest used in defining the load for the adjoint
-   * problem with respect to the shape displacement field
+   * problem with respect to the shape displacement field for the last computed adjoint timestep
    *
    * @return The sensitivity with respect to the shape displacement
    *
-   * @pre `solveAdjoint` with an appropriate adjoint load must be called prior to this method.
+   * @pre `reverseAdjointTimestep` with an appropriate adjoint load must be called prior to this method.
    */
-  FiniteElementDual& computeShapeSensitivity() override
+  FiniteElementDual& computeTimestepShapeSensitivity() override
   {
-    auto drdshape = serac::get<DERIVATIVE>((*residual_)(DifferentiateWRT<SHAPE>{}, temperature_, zero_,
+    auto drdshape     = serac::get<DERIVATIVE>((*residual_)(DifferentiateWRT<SHAPE>{}, temperature_, temperature_rate_,
                                                         shape_displacement_, *parameters_[parameter_indices].state...));
-
     auto drdshape_mat = assemble(drdshape);
 
     drdshape_mat->MultTranspose(adjoint_temperature_, shape_displacement_sensitivity_);
 
     return shape_displacement_sensitivity_;
   }
 
+  /**
+   * @brief Compute the implicit sensitivity of the quantity of interest with respect to the initial temperature
+   *
+   * @return The sensitivity with respect to the initial temperature
+   *
+   * @pre `reverseAdjointTimestep` must be called as many times as the forward solver was advanced before this is called
+   */
+  const std::unordered_map<std::string, const serac::FiniteElementDual&> computeInitialConditionSensitivity() override
```

Comment:
As this is technically available at any point during the reverse timestep process, is it worth documenting what it is when accessed in that state? @sethwatts , do you think you would ever use it?

Reply:
If the question is, would we use this method, then yes, I could definitely see doing inverse problems where we are trying to deduce initial conditions, of find optimal ones t evolve to a desired end state. If the question is, would you ever try using it before working to the end adjoint time / zero primal time, I think the answer is no.

Reply:
Before getting back to time zero, that quantity would be the implicit sensitivity of the qoi with respect to the temperature at the start of the most recent cycle/timestep.  Not something one would commonly be interested in, but it does have an interpretation.

Reply:
Does it need its own member function then, or is it equivalent to just computing sensitivities after returning to cycle zero in the adjoint pass?

Reply:
> Not something one would commonly be interested in, but it does have an interpretation.

Agreed on both points.

Reply:
In principal, it could be re-computed with the 'adjoint temperature', and multiplying it by just the mass matrix transposed at reverse time=0.  However, this would force the user to essentially rederive (and recompute) a quantity that we already store.  And this multiply would presumably require an new API function anyway.  In my view, I think that breaks encapsulation too much to justify (potentially) reducing the number of public API functions.

Reply:
Also, I'm voting for not documenting what it is if accessed early.  This way we do not have to commit to it being available for all physics before the end of the adjoint pass? There could be future optimizations that take advantage of this flexibility.